client/src/components/AddLatex/index.js

```diff
@@ -0,0 +1,188 @@
+import Button from "react-bootstrap/Button";
+import Dropdown from "react-bootstrap/Dropdown";
+import AsyncSelect from "react-select/async";
+import makeAnimated from "react-select/animated";
+
+import API from "../../utils/API";
+const animatedComponent = makeAnimated();
+
+class AddLatexForm extends Component {
+    state = {
+        selectedUsers: [],
+        validated: false,
+        collectionDate: "2022-02-02",
+        tareWeight: 0,
+        grossWeight: 1,
+        barrelNumber: "",
+    };
+    onChange = (selectedUsers) => {
+        this.setState({
+            selectedUsers: selectedUsers || [],
+        });
+        console.log(selectedUsers);
+    };
+
+    loadOptions = async (inputText, callback) => {
+        const response = await fetch(
+            `https://nalandaapi.herokuapp.com/api/newcustomer/search?searchString=${inputText}`
+        );
+        const json = await response.json();
+
+        callback(json.map((i) => ({ label: i.customerName, value: i.customerId })));
+    };
+
+    // Handles updating component state when the user types into the input field
+    handleInputChange = (event) => {
+        const { name, value } = event.target;
+
+        this.setState({
+            [name]: value,
+        });
+    };
+
+    handleOnSubmit = (event) => {
+        event.preventDefault();
+        const form = event.currentTarget;
+        console.log("Submited");
+        if (form.checkValidity() === false) {
+            event.preventDefault();
+            event.stopPropagation();
+            this.setState({ validated: true });
+        } else {
+            this.setState({ validated: true });
+            const newLatexEntry = {
+                customerId: this.state.selectedUsers[0].value,
+                collectionDate: this.state.collectionDate,
+                tareWeight: this.state.tareWeight,
+                grossWeight: this.state.grossWeight,
+                barrelNumber: this.state.barrelNumber,
+            };
+            console.log(newLatexEntry);
+            API.addLatexEntry(newLatexEntry)
+                .then((res) => {
+                    console.log(res);
+                    this.setState({
+                        selectedUsers: [],
+                        validated: false,
+                        collectionDate: "2022-02-02",
+                        tareWeight: 0,
+                        grossWeight: 1,
+                        barrelNumber:"",
+                    });
+                    this.props.closeAddlatexForm();
+                })
+                .catch((err) => {
+                    console.log(err);
+                });
+        }
+    };
+
+    render() {
+        return this.props.trigger ? (
+            <div className="newLatex">
+                {" "}
+                <div className="newLatexInner">
+                    <h1>Add Latex Collection</h1>
+                    <br></br>
+                    <AsyncSelect
+                        isMulti
```

Comment:
do we need isMulti here?

Reply:
No sir even if we select multiple customers and add collection it reflects only for one.

Reply:
I will remove it and will make a new PR after making every other correction